pkg/reconciler/ingress/resources/virtual_service.go

```diff
@@ -219,6 +219,10 @@ func makeVirtualServiceRoute(hosts sets.String, http *v1alpha1.HTTPIngressPath,
 		if http.Retries.PerTryTimeout != nil {
 			route.Retries.PerTryTimeout = types.DurationProto(http.Retries.PerTryTimeout.Duration)
 		}
+	} else {
+		route.Retries = &istiov1alpha3.HTTPRetry{
+			Attempts: 0,
+		}
```

Comment:
if you want to codegolf even more you can just do the following:
```
route.Retries = &istiov1alpha3.HTTPRetry{} // Attempts: 0 is automatic
if http.Retries != nil && ... {
  route.Retries.Retryon = 
  route.Retries.Attempts = ...
  if .. {
  }
}
```

Reply:
Sure, looks better. Updated.